fendermint/vm/actor_interface/src/diamond.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! Helper data structures to declare diamond pattern contracts.
+
+// See https://medium.com/@MarqyMarq/how-to-implement-the-diamond-standard-69e87dae44e6
+
+use std::collections::HashMap;
+
+use ethers::abi::Abi;
+use fvm_shared::ActorID;
+
+#[derive(Clone, Debug)]
+pub struct EthFacet {
+    pub name: &'static str,
```

Comment:
Noob Rust question, why use a `'static` here?

Reply:
I could use `String` which would be owned by the `EthFacet` and could be anything, and I'd have to declare it like `name: "Gateway".to_owned(), ` or `.into()` or `.to_string()`, or I can use what you normally see as `&str` and declare it like `name: "Gateway", `, but in that case because this isn't a parameter to a function, I have to say what the lifetime of this reference is, and `&'static` means the lifetime is that of the entire program. This means that these are statically defined names typed into the editor, not coming from a file or something. 

We can change this if we have to, even use macros to construct this thing to have a nice DSL, but this was my first idea: that you have to describe in code what these contract's look like, like you do in your deployment scripts, so `&'static` is fine.